metadata-ingestion/src/datahub/ingestion/source/grafana/grafana_source.py

```diff
@@ -1,131 +1,469 @@
+                fields=[
+                    InputFieldClass(
+                        schemaField=field,
+                        schemaFieldUrn=make_schema_field_urn(
+                            dataset_urn, field.fieldPath
                         ),
-                        StatusClass(removed=False),
-                    ],
-                )
+                    )
+                    for field in input_fields
+                ]
+            ),
+        ).as_workunit()
+
+    def _process_panel_dataset(
+        self, panel: Panel, dashboard_uid: str, ingest_tags: bool
+    ) -> Iterable[MetadataWorkUnit]:
+        """Process dataset metadata for a panel"""
+        if not panel.datasource:
+            return
+
+        ds_type = panel.datasource.get("type", "unknown")
+        ds_uid = panel.datasource.get("uid", "unknown")
+
+        # Build dataset name
+        dataset_name = f"{ds_type}.{ds_uid}.{panel.id}"
+
+        # Create dataset URN
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            platform=self.platform,
+            name=dataset_name,
+            platform_instance=self.platform_instance,
+            env=self.env,
+        )
+
+        # Create dataset snapshot
+        dataset_snapshot = DatasetSnapshotClass(
+            urn=dataset_urn,
+            aspects=[
+                DataPlatformInstanceClass(
+                    platform=make_data_platform_urn(self.platform),
+                    instance=make_dataplatform_instance_urn(
+                        platform=self.platform,
+                        instance=self.platform_instance,
+                    )
+                    if self.platform_instance
+                    else None,
+                ),
+                DatasetPropertiesClass(
+                    name=f"{ds_uid} ({panel.title or panel.id})",
+                    description="",
+                    customProperties={
+                        "type": ds_type,
+                        "uid": ds_uid,
+                        "full_path": dataset_name,
+                    },
+                ),
+                StatusClass(removed=False),
+            ],
+        )
+
+        # Add schema metadata if available
+        schema_fields = extract_fields_from_panel(panel)
+        if schema_fields:
+            schema_metadata = SchemaMetadataClass(
+                schemaName=f"{ds_type}.{ds_uid}.{panel.id}",
+                platform=make_data_platform_urn(self.platform),
+                version=0,
+                fields=schema_fields,
+                hash="",
+                platformSchema=OtherSchemaClass(rawSchema=""),
             )
+            dataset_snapshot.aspects.append(schema_metadata)
+
+        if dashboard_uid and self.config.ingest_tags:
+            dashboard = self.client.get_dashboard(dashboard_uid)
+            if dashboard and dashboard.tags:
+                tags = []
+                for tag in dashboard.tags:
+                    if ":" in tag:
```

Comment:
I'm not sure whether this is necessary  - we have been using tags in form `key:value` in other sources such as s3, bigquery.